forum/startups/tests.py

```diff
@@ -217,5 +221,109 @@ def test_profile_creation(self):
         self.assertEqual(self.user.username, "defaultuser")
+        self.assertIn(self.industry1, profile.industries.all())
+
+
+class StartupViewSetTests(APITestCase, StartupTestBase):
+    """
+    Unit tests for the StartupViewSet functionality.
+    This test class verifies the retrieval and listing of startup profiles using 
+    Django's APITestCase to simulate authenticated requests and data retrieval.
+
+    Methods:
+        setUp(): Initializes test data, including a user with 'investor' role, 
+        startup profiles.
+    
+        test_retrieve_startup_success(): Tests retrieving an existing startup profile
+        by sending a GET request to the detail view endpoint and checking the response 
+        status and data accuracy.
+    
+        test_retrieve_startup_not_found(): Tests retrieving a non-existent startup profile 
+        by sending a GET request with a random UUID, expecting a 404 response.
+
+        test_list_startups_success(): Tests listing all available startup profiles by 
+        sending a GET request to the list view endpoint and validating the response data.
+
+        test_search_startup_by_company_name(): Tests searching startup profiles by a 
+        partial company name by sending a GET request with a search query parameter.
+    
+        test_filter_startup_by_exact_company_name(): Tests filtering startup profiles 
+        by an exact company name by sending a GET request with a company_name query 
+        parameter.
+    
+        test_filter_startup_by_industry_name(): Tests filtering startup profiles based on 
+        industry by sending a GET request with an industry_name query parameter. 
+"""
+    def setUp(self):
+
+        self.user, self.industry1, self.location = self.create_common_test_data()
+        self.industry2 = Industry.objects.create(name='finance')
+        investor_role = Role.objects.create(name='investor')
+
+        self.user.change_active_role('investor')
+        self.client.force_authenticate(user=self.user)
+
+        self.startup = Startup.objects.create(
+            user_id=self.user.user_id,
+            company_name="EcoSolutionss",
+            required_funding=7150000.00,
+            funding_stage=FundingStage.SEED,
+            number_of_employees=30,
+            location=self.location,
+            description="Eco-friendly solutions provider",
+            total_funding=20000.00,
+            website="https://www.ecosolutions.com",
+        ) 
+        self.startup2 = Startup.objects.create(
+            user_id=self.user.user_id,
+            company_name="Solar",
+            required_funding=1000000,
+            location=self.location,
+        )
+        self.startup.industries.set([self.industry1])
+        self.startup2.industries.set([self.industry2])
+
+        self.list_url = reverse('startup-list')
+        self.retrieve_url = reverse('startup-detail', args=[self.startup.startup_id])
```

Comment:
After creating startup instances, include assertions to confirm these objects were created successfully. This will ensure `setUp` ran as expected and help with debugging if an error occurs here.
```
def setUp(self):
    # Existing setup code...
    self.assertTrue(Startup.objects.filter(company_name="EcoSolutionss").exists())
    self.assertTrue(Startup.objects.filter(company_name="Solar").exists())
```